src/NaNMath.jl

```diff
@@ -61,6 +61,48 @@ function sum(x::AbstractArray{T}) where T<:AbstractFloat
     return result
 end
 
+"""
+NaNMath.median(A)
+
+##### Args:
+* `A`: An array of floating point numbers
+
+##### Returns:
+*   Returns the median of all elements in the array, ignoring NaN's.
+    Returns NaN for an empty array or array containing NaNs only.
+
+##### Examples:
+```julia
+using NaNMath
+NaNMath.median([1., 2., 3., NaN]) # result: 2.
+NaNMath.median([1., 2., NaN]) # result: 1.5
+NaNMath.median([NaN]) # result: NaN
+```
+"""
+function median(x::AbstractArray{T}) where T<:AbstractFloat
+
+    if ndims(x) > 1
+        x = collect(Iterators.flatten(x))
+    end
+
+    filter!(el->!isnan(el), x)
+    sort!(x)
```

Comment:
This mutuates the input array `x`, which we do not want

```julia
julia> x = [1, NaN, 3];

julia> median(x);

julia> x
2-element Array{Float64,1}:
 1.0
 3.0
```

We can instead use non-mutating versions `filter` and `sort` i.e.
```julia
    x = sort(filter(!isnan, x))
```

Reply:
Good catch, forgot the `!` is only a convention